diff --git a/source/TrampolineBridge/ClosureTrampolineBridge/arm64/closure_bridge_arm64.asm b/source/TrampolineBridge/ClosureTrampolineBridge/arm64/closure_bridge_arm64.asm
--- a/source/TrampolineBridge/ClosureTrampolineBridge/arm64/closure_bridge_arm64.asm
+++ b/source/TrampolineBridge/ClosureTrampolineBridge/arm64/closure_bridge_arm64.asm
@@ -51,8 +51,13 @@ str TMP_REG_0, [sp, #(1 * 8)]
 // call convention: x0 = register context, x1 = interceptor entry
 mov x0, sp
 ldr x1, [sp, #(2 * 8 + 2 * 8 + 30 * 8 + 8 * 16)]
+#if defined(__APPLE__)
 adrp TMP_REG_0, cdecl(common_closure_bridge_handler)@PAGE
 add TMP_REG_0, TMP_REG_0, cdecl(common_closure_bridge_handler)@PAGEOFF
+#else
+adrp TMP_REG_0, cdecl(common_closure_bridge_handler)
+add TMP_REG_0, TMP_REG_0, :lo12:cdecl(common_closure_bridge_handler)
+#endif
 blr TMP_REG_0
 
 // restore stack, saved original sp
diff --git a/cmake/compiler_and_linker.cmake b/cmake/compiler_and_linker.cmake
--- a/cmake/compiler_and_linker.cmake
+++ b/cmake/compiler_and_linker.cmake
@@ -44,7 +44,11 @@ endif ()
 endif ()
 
 if (PROCESSOR.ARM)
-  set(CMAKE_ASM_FLAGS "${CMAKE_ASM_FLAGS} -arch armv7 -x assembler-with-cpp")
+  if (SYSTEM.Android)
+    set(CMAKE_ASM_FLAGS "${CMAKE_ASM_FLAGS} -x assembler-with-cpp")
+  else ()
+    set(CMAKE_ASM_FLAGS "${CMAKE_ASM_FLAGS} -arch armv7 -x assembler-with-cpp")
+  endif ()
 elseif (PROCESSOR.AARCH64)
   set(CMAKE_ASM_FLAGS "${CMAKE_ASM_FLAGS} -x assembler-with-cpp")
 endif ()
